### PR TITLE
Add reSync task callback

### PR DESCRIPTION
**Pull request by Monokaix**
Related issus: https://github.com/volcano-sh/volcano/issues/3096

### CODE REVIEW

File: pkg/scheduler/cache/cache.go [hunk 1]
```diff
@@ -961,8 +961,18 @@ func (sc *SchedulerCache) processResyncTask() {
 		return
 	}
 
+	reSync := false
 	if err := sc.syncTask(task); err != nil {
 		klog.Errorf("Failed to sync pod <%v/%v>, retry it.", task.Namespace, task.Name)
+		reSync = true
+	}
+	if task.ReSyncTask != nil {
```
Comment (lowang-bh) [inline pkg/scheduler/cache/cache.go:969]: How about syncTask return no error? Looks it will always run ReSyncTask callback.

Comment (Monokaix) [reply pkg/scheduler/cache/cache.go:969]: yes, `ReSyncTask callback` will always retry and run until it return no err.

Comment (Monokaix) [reply pkg/scheduler/cache/cache.go:969]: I think what we need care is `syncTask` return err but `ReSyncTask ` return no err, then `ReSyncTask` will still run every time, we can add a flag to indicate whether `ReSyncTask` is finished.

Comment (lowang-bh) [reply pkg/scheduler/cache/cache.go]: Which scenario will this callback be used in?

Comment (Monokaix) [reply pkg/scheduler/cache/cache.go]: just like what I said in description, when using custom bind method, some alarm or rollback are needed when bind err.

File: pkg/scheduler/api/job_info.go [hunk 2]
```diff
@@ -130,6 +131,10 @@ type TaskInfo struct {
 	NumaInfo   *TopologyInfo
 	PodVolumes *volumescheduling.PodVolumes
 	Pod        *v1.Pod
+
+	// CustomBindErrHandler is a custom callback func called when task bind err.
+	CustomBindErrHandler         func() error
+	CustomBindErrHandlerFinished bool
```
Comment (william-wang) [inline pkg/scheduler/api/job_info.go:137]: please change `CustomBindErrHandlerFinished` to a better name.

Comment (Monokaix) [reply pkg/scheduler/api/job_info.go]: done for that.

**Review by william-wang — APPROVED** [review]
/approve

**Pull request merged**